connection.go

```diff
@@ -279,3 +275,26 @@ func (conn *Connection) checkTimeout() error {
 		conn.Unlock()
 	}
 }
+
+// Update exponential smoothing throughput
+func (conn *Connection) updateEtp() {
+	floatTp := float64(conn.tp)
+	if conn.etp == 0 {
+		conn.etp = floatTp
+	} else {
+		conn.etp = alpha*floatTp + (1.0-alpha)*(conn.etp)
+	}
+
+	conn.tp = 0 // reset it for next time
+}
+
+func (conn *Connection) setWindowSize(n uint32) {
+	if n == 0 {
+		return
+	}
+	if n < uint32(conn.session.config.MinConnectionWindowSize) {
```

Comment:
What about MaxConnectionWindowSize? We can change the default value to 0 and make 0 means unlimited